web/src/pages/billing/BillingCostByMonth.tsx

```diff
@@ -164,23 +181,27 @@ const BillingCostByTime: React.FunctionComponent = () => {
         )
     }
 
+    const onMonthStart = (event: any, data: any) => {
+        changeDate('start', data.value)
+    }
+
+    const onMonthEnd = (event: any, data: any) => {
+        changeDate('end', data.value)
+    }
+
     React.useEffect(() => {
-        if (
-            start !== undefined &&
-            start !== '' &&
-            start !== null &&
-            end !== undefined &&
-            end !== '' &&
-            end !== null
-        ) {
+        if (Boolean(start) && Boolean(end)) {
             // valid selection, retrieve data
             getData({
                 fields: [BillingColumn.Topic, BillingColumn.CostCategory],
-                start_date: start,
-                end_date: end,
+                start_date: getAdjustedDay(convertInvoiceMonth(start, true), -2),
+                end_date: getAdjustedDay(convertInvoiceMonth(end, false), 2),
```

Comment:
For good measure haha
```suggestion
                end_date: getAdjustedDay(convertInvoiceMonth(end, false), 3),
```